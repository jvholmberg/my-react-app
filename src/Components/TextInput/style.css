.TextInput-Container {
  padding: .5rem 0;
}
.TextInput-Label {
  font-weight: 300;
}
.TextInput-Required {
  color: #dc3545;
}
.TextInput {
  display: block;
  margin-top: .5rem;
  padding: .5rem 1rem;
  width: 100%;
  font-size: 1rem;
  font-weight: 300;
  box-sizing: border-box;
}
.TextInput.Primary {
  border: none;
  background-color: #f1f1f1;
  border-bottom: .1rem solid #007bff;
}
.TextInput.Secondary {
  border: none;
  background-color: #f1f1f1;
  border-bottom: .1rem solid #6c757d;
}
.TextInput.Success {
  border: none;
  background-color: #f1f1f1;
  border-bottom: .1rem solid #28a745;
}
.TextInput.Danger {
  border: none;
  background-color: #f1f1f1;
  border-bottom: .1rem solid #dc3545;
}
.TextInput.Warning {
  border: none;
  background-color: #f1f1f1;
  border-bottom: .1rem solid #ffc107;
}
.TextInput.Info {
  border: none;
  background-color: #f1f1f1;
  border-bottom: .1rem solid #17a2b8;
}
.TextInput.Light {
  background-color: #f1f1f1;
  border: none;
}
.TextInput.Dark {
  background-color: #bcbcbd;
  border: none;
}
