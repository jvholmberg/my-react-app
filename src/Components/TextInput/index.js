import React from 'react';
import { CraftClassName } from '../';

import './style.css';

const TextInput = ({
  name, label, placeholder, password, required,
  // All touched inputs for the entire form, get by name
  touched,
  // Validation errors for the entire form, get by name
  errors,
  // Other props passed
  ...props,
}) => {
  const type = {
    primary: touched && !touched[name],
    success: errors && !errors[name],
    danger: errors && touched && errors[name] && touched[name],
  };
  let className = CraftClassName(type, 'TextInput', 'Primary');
  return (
    <div className='TextInput-Container'>
      <label className='TextInput-Label'>
        {label} {required && (<span className='TextInput-Required'>*</span>)}
        <input
          name={name}
          placeholder={placeholder}
          type={password ? 'password' : 'text'}
          className={className}
          {...props} />
      </label>
    </div>
  );
}
export default TextInput;
