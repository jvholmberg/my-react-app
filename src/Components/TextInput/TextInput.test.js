import React from 'react';

import TextInput from './';

describe('TextInput', () => {
  it('Empty matches snapshot', () => {
    expect(shallow(
      <TextInput />
    )).toMatchSnapshot();
  });
  it('With label matches snapshot', () => {
    expect(shallow(
      <TextInput label='label' />
    )).toMatchSnapshot();
  });
  it('Danger matches snapshot', () => {
    expect(shallow(
      <TextInput danger />
    )).toMatchSnapshot();
  });
  it('Typical censored matches snapshot', () => {
    expect(shallow(
      <TextInput required />
    )).toMatchSnapshot();
  });
  it('Typical censored with className matches snapshot', () => {
    expect(shallow(
      <TextInput className='className' required />
    )).toMatchSnapshot();
  });
});
