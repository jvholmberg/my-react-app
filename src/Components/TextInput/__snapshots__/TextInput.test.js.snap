// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput Danger matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TextInput
    danger={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <label
        className="TextInput-Label"
>
         
        <input
                className="TextInput Primary"
                danger={true}
                name={undefined}
                placeholder={undefined}
                type="text"
        />
</label>,
      "className": "TextInput-Container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          undefined,
          " ",
          undefined,
          <input
            className="TextInput Primary"
            danger={true}
            name={undefined}
            placeholder={undefined}
            type="text"
/>,
        ],
        "className": "TextInput-Label",
      },
      "ref": null,
      "rendered": Array [
        undefined,
        " ",
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "TextInput Primary",
            "danger": true,
            "name": undefined,
            "placeholder": undefined,
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "label",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <label
          className="TextInput-Label"
>
           
          <input
                    className="TextInput Primary"
                    danger={true}
                    name={undefined}
                    placeholder={undefined}
                    type="text"
          />
</label>,
        "className": "TextInput-Container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            undefined,
            " ",
            undefined,
            <input
              className="TextInput Primary"
              danger={true}
              name={undefined}
              placeholder={undefined}
              type="text"
/>,
          ],
          "className": "TextInput-Label",
        },
        "ref": null,
        "rendered": Array [
          undefined,
          " ",
          undefined,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "TextInput Primary",
              "danger": true,
              "name": undefined,
              "placeholder": undefined,
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "label",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`TextInput Empty matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TextInput />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <label
        className="TextInput-Label"
>
         
        <input
                className="TextInput Primary"
                name={undefined}
                placeholder={undefined}
                type="text"
        />
</label>,
      "className": "TextInput-Container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          undefined,
          " ",
          undefined,
          <input
            className="TextInput Primary"
            name={undefined}
            placeholder={undefined}
            type="text"
/>,
        ],
        "className": "TextInput-Label",
      },
      "ref": null,
      "rendered": Array [
        undefined,
        " ",
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "TextInput Primary",
            "name": undefined,
            "placeholder": undefined,
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "label",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <label
          className="TextInput-Label"
>
           
          <input
                    className="TextInput Primary"
                    name={undefined}
                    placeholder={undefined}
                    type="text"
          />
</label>,
        "className": "TextInput-Container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            undefined,
            " ",
            undefined,
            <input
              className="TextInput Primary"
              name={undefined}
              placeholder={undefined}
              type="text"
/>,
          ],
          "className": "TextInput-Label",
        },
        "ref": null,
        "rendered": Array [
          undefined,
          " ",
          undefined,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "TextInput Primary",
              "name": undefined,
              "placeholder": undefined,
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "label",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`TextInput Typical censored matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TextInput
    required={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <label
        className="TextInput-Label"
>
         
        <span
                className="TextInput-Required"
        >
                *
        </span>
        <input
                className="TextInput Primary"
                name={undefined}
                placeholder={undefined}
                type="text"
        />
</label>,
      "className": "TextInput-Container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          undefined,
          " ",
          <span
            className="TextInput-Required"
>
            *
</span>,
          <input
            className="TextInput Primary"
            name={undefined}
            placeholder={undefined}
            type="text"
/>,
        ],
        "className": "TextInput-Label",
      },
      "ref": null,
      "rendered": Array [
        undefined,
        " ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "*",
            "className": "TextInput-Required",
          },
          "ref": null,
          "rendered": "*",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "TextInput Primary",
            "name": undefined,
            "placeholder": undefined,
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "label",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <label
          className="TextInput-Label"
>
           
          <span
                    className="TextInput-Required"
          >
                    *
          </span>
          <input
                    className="TextInput Primary"
                    name={undefined}
                    placeholder={undefined}
                    type="text"
          />
</label>,
        "className": "TextInput-Container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            undefined,
            " ",
            <span
              className="TextInput-Required"
>
              *
</span>,
            <input
              className="TextInput Primary"
              name={undefined}
              placeholder={undefined}
              type="text"
/>,
          ],
          "className": "TextInput-Label",
        },
        "ref": null,
        "rendered": Array [
          undefined,
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "*",
              "className": "TextInput-Required",
            },
            "ref": null,
            "rendered": "*",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "TextInput Primary",
              "name": undefined,
              "placeholder": undefined,
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "label",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`TextInput Typical censored with className matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TextInput
    className="className"
    required={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <label
        className="TextInput-Label"
>
         
        <span
                className="TextInput-Required"
        >
                *
        </span>
        <input
                className="className"
                name={undefined}
                placeholder={undefined}
                type="text"
        />
</label>,
      "className": "TextInput-Container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          undefined,
          " ",
          <span
            className="TextInput-Required"
>
            *
</span>,
          <input
            className="className"
            name={undefined}
            placeholder={undefined}
            type="text"
/>,
        ],
        "className": "TextInput-Label",
      },
      "ref": null,
      "rendered": Array [
        undefined,
        " ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "*",
            "className": "TextInput-Required",
          },
          "ref": null,
          "rendered": "*",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "className",
            "name": undefined,
            "placeholder": undefined,
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "label",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <label
          className="TextInput-Label"
>
           
          <span
                    className="TextInput-Required"
          >
                    *
          </span>
          <input
                    className="className"
                    name={undefined}
                    placeholder={undefined}
                    type="text"
          />
</label>,
        "className": "TextInput-Container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            undefined,
            " ",
            <span
              className="TextInput-Required"
>
              *
</span>,
            <input
              className="className"
              name={undefined}
              placeholder={undefined}
              type="text"
/>,
          ],
          "className": "TextInput-Label",
        },
        "ref": null,
        "rendered": Array [
          undefined,
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "*",
              "className": "TextInput-Required",
            },
            "ref": null,
            "rendered": "*",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "className",
              "name": undefined,
              "placeholder": undefined,
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "label",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`TextInput With label matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TextInput
    label="label"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <label
        className="TextInput-Label"
>
        label
         
        <input
                className="TextInput Primary"
                name={undefined}
                placeholder={undefined}
                type="text"
        />
</label>,
      "className": "TextInput-Container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "label",
          " ",
          undefined,
          <input
            className="TextInput Primary"
            name={undefined}
            placeholder={undefined}
            type="text"
/>,
        ],
        "className": "TextInput-Label",
      },
      "ref": null,
      "rendered": Array [
        "label",
        " ",
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "TextInput Primary",
            "name": undefined,
            "placeholder": undefined,
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "label",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <label
          className="TextInput-Label"
>
          label
           
          <input
                    className="TextInput Primary"
                    name={undefined}
                    placeholder={undefined}
                    type="text"
          />
</label>,
        "className": "TextInput-Container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "label",
            " ",
            undefined,
            <input
              className="TextInput Primary"
              name={undefined}
              placeholder={undefined}
              type="text"
/>,
          ],
          "className": "TextInput-Label",
        },
        "ref": null,
        "rendered": Array [
          "label",
          " ",
          undefined,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "TextInput Primary",
              "name": undefined,
              "placeholder": undefined,
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "label",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
